firewood/src/merkle.rs

```diff
@@ -386,10 +390,9 @@ impl<T: WriteLinearStore> Merkle<T> {
                     partial_path: Path::from(old_root_partial_path),
                 }),
                 Node::Branch(old_root) => Node::Branch(Box::new(BranchNode {
-                    children: old_root.children,
+                    children: old_root.children.clone(),
```

Comment:
> Was the type Copy before because the address is Copy?

Yes

> It doesn't look like old_root is used after this, can't you avoid the borrow at the start of this match statement?

I don't think we can move out of `old_root` (the first declaration) because it's an Arc